### PR TITLE
[SYCL] Use teams to assign code ownership

### PR DESCRIPTION
**Pull request by vladimirlaz**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,110 +1,77 @@
...
-sycl/include/sycl/ext/intel/reduction.hpp @v-klochkov
-
-# Function pointers
-sycl/include/sycl/ext/intel/function_pointer.hpp @AlexeySachkov
-sycl/source/function_pointer.cpp @AlexeySachkov
-
-# Half Type
-sycl/include/CL/sycl/half_type.hpp @AlexeySachkov
-sycl/source/half_type.cpp @AlexeySachkov
-
```
Comment (vladimirlaz) [inline .github/CODEOWNERS:76]: these files will fallback to @intel/llvm-reviewers-runtime

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,110 +1,77 @@
...
+sycl/ReleaseNotes.md @dpcpp-doc-reviewers
+sycl/doc/ @dpcpp-doc-reviewers
 sycl/doc/extensions/ @intel/dpcpp-specification-reviewers
-sycl/doc/extensions/SPIRV/ @AlexeySotkin @bashbaug @mbelicki
-
-# Sub-groups
-sycl/include/CL/sycl/detail/spirv.hpp @Pennycook @AlexeySachkov
-sycl/include/sycl/ext/intel/group_algorithm.hpp @Pennycook @AlexeySachkov
-sycl/include/sycl/ext/intel/sub_group.hpp @Pennycook @AlexeySachkov
-sycl/include/sycl/ext/intel/sub_group_host.hpp @Pennycook @AlexeySachkov
```
Comment (vladimirlaz) [inline .github/CODEOWNERS:34]: these files will fallback to @intel/llvm-reviewers-runtime

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,110 +1,77 @@
...
+clang/tools/clang-offload-wrapper/ @intel/dpcpp-tools-reviewers
+clang/tools/clang-offload-deps/ @intel/dpcpp-tools-reviewers
+clang/tools/clang-offload-extract/ @intel/dpcpp-tools-reviewers
 
 # Explicit SIMD
-SYCLLowerIR/ @kbobrovs @sndmitriev @kychendev @v-klochkov 
-esimd/ @kbobrovs @v-klochkov @kychendev
-sycl/include/sycl/ext/intel/experimental/esimd.hpp @kbobrovs @v-klochkov @kychendev
-sycl/doc/extensions/ExplicitSIMD/ @kbobrovs @v-klochkov @kychendev
+SYCLLowerIR/ @inteldpcpp-esimd-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:69]:
```suggestion
SYCLLowerIR/ @intel/dpcpp-esimd-reviewers
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,110 +1,77 @@
...
 # Front-end compiler
-clang/ @premanandrao @elizabethandrews @smanna12
-
-# Driver
-clang/**/Driver @mdtoguchi @AGindinson @hchilama
+clang/ @dpcpp-cfe-reviewers
 
 # LLVM-SPIRV translator
-llvm-spirv/ @AlexeySotkin @AlexeySachkov
+llvm-spirv/ @intel/dpcpp-spirv-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:7]: Do we have all the teams referenced in this file?

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -1,110 +1,77 @@
...
-libdevice/ @vzakhari @jinge90
+libdevice/ @intel/llvm-reviewers-runtime
 
 # DPC++ runtime library
 sycl/ @intel/llvm-reviewers-runtime
 
 # Documentation
-sycl/ReleaseNotes.md @pvchupin @tfzhu
-sycl/doc/ @pvchupin @bader
+sycl/ReleaseNotes.md @dpcpp-doc-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:22]:
```suggestion
sycl/ReleaseNotes.md @intel/dpcpp-doc-reviewers
```

**Comment by bader** [comment]
@vladimirlaz, please, use a private fork to open a pull request.

File: .github/CODEOWNERS (hunk 4 above)
Comment (vladimirlaz) [reply .github/CODEOWNERS:7]: We will have to create them once all prior code owners agreed on the change. Will ping them soon.

File: .github/CODEOWNERS (hunk 5 above)
Comment (bader) [reply .github/CODEOWNERS]: Please, fix all instances.

File: .github/CODEOWNERS (hunk 3 above)
Comment (bader) [reply .github/CODEOWNERS]: Please, fix all instances.

**Review by AlexeySotkin — DISMISSED** [review]
SPIR-V part looks good to me.

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -1,110 +1,77 @@
...
-opencl/ @pvchupin @vladimirlaz
+opencl/ @intel/llvm-reviewers-runtime
 
 # OpenCL "offline" compiler
-opencl/opencl-aot/ @dm-vodopyanov @AlexeySachkov @romanovvlad
+opencl/opencl-aot/ @intel/dpcpp-tools-reviewers
 
 # Device library
-libdevice/ @vzakhari @jinge90
+libdevice/ @intel/llvm-reviewers-runtime
```
Comment (romanovvlad) [inline .github/CODEOWNERS:16]: It seems it becomes a part of SYCL RT.

**Comment by bader** [comment]
@vladimirlaz, ping.

File: .github/CODEOWNERS [hunk 7]
```diff
@@ -1,110 +1,73 @@
...
+llvm/**/llvm-foreach/ @intel/dpcpp-tools-reviewers
+llvm/**/llvm-no-spir-kernel/ @intel/dpcpp-tools-reviewers
+llvm/**/sycl-post-link/ @intel/dpcpp-tools-reviewers
+llvm/include/llvm/Support/PropertySetIO.h @intel/dpcpp-tools-reviewers
+llvm/lib/Support/PropertySetIO.cpp @intel/dpcpp-tools-reviewers
+llvm/unittests/Support/PropertySetIOTest.cpp @intel/dpcpp-tools-reviewers
+llvm/include/llvm/Support/SimpleTable.h @intel/dpcpp-tools-reviewers
+llvm/lib/Support/SimpleTable.cpp @intel/dpcpp-tools-reviewers
+llvm/unittests/Support/SimpleTableTest.cpp @intel/dpcpp-tools-reviewers
+llvm/lib/Support/Base64.cpp @intel/dpcpp-tools-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:56]:
```suggestion
llvm/ @intel/dpcpp-tools-reviewers
```

File: .github/CODEOWNERS [hunk 8]
```diff
@@ -1,110 +1,73 @@
...
 
 # Clang offload tools
-clang/tools/clang-offload-bundler/ @mlychkov @sndmitriev @AlexeySachkov
-clang/tools/clang-offload-wrapper/ @mlychkov @sndmitriev @AlexeySachkov @kbobrovs
-clang/tools/clang-offload-deps/ @sndmitriev @mlychkov @AlexeySachkov
-clang/tools/clang-offload-extract/ @sndmitriev @mlychkov @AlexeySachkov
+clang/tools/clang-offload-bundler/ @intel/dpcpp-tools-reviewers
+clang/tools/clang-offload-wrapper/ @intel/dpcpp-tools-reviewers
+clang/tools/clang-offload-deps/ @intel/dpcpp-tools-reviewers
+clang/tools/clang-offload-extract/ @intel/dpcpp-tools-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:62]:
```suggestion
clang/tools/clang-offload-*/ @intel/dpcpp-tools-reviewers
```
How about matching all clang tools named with `clang-offload-` prefix.

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -1,110 +1,73 @@
...
-SYCLLowerIR/ @kbobrovs @sndmitriev @kychendev @v-klochkov 
-esimd/ @kbobrovs @v-klochkov @kychendev
-sycl/include/sycl/ext/intel/experimental/esimd.hpp @kbobrovs @v-klochkov @kychendev
-sycl/doc/extensions/ExplicitSIMD/ @kbobrovs @v-klochkov @kychendev
+SYCLLowerIR/ @intel/dpcpp-esimd-reviewers
+esimd/ @kbobrovs @intel/dpcpp-esimd-reviewers
+sycl/include/sycl/ext/intel/experimental/esimd.hpp @intel/dpcpp-esimd-reviewers
+sycl/doc/extensions/ExplicitSIMD/ @intel/dpcpp-esimd-reviewers
 
 # ITT annotations
```
Comment (bader) [inline .github/CODEOWNERS:70]: We can remove this section if @intel/dpcpp-tools-reviewers team okay to review changes in llvm/ directory.

**Review by bader — COMMENTED** [review]
LGTM, a couple of suggestions, which hopefully are okay.

**Comment by bader** [comment]
@vladimirlaz, is there an infrastructure team we can assign as a code owner for .github script sources?

**Comment by bader** [comment]
@vladimirlaz, please, resolve merge conflicts.

**Review by bader — CHANGES_REQUESTED** [review]
Please, fix pre-commit checks.

**Comment by alexbatashev** [comment]
@vladimirlaz @bader https://github.com/intel/llvm-test-suite/pull/684

**Review by bader — APPROVED** [review]

File: .github/CODEOWNERS [hunk 10]
```diff
@@ -1,116 +1,55 @@
...
-sycl/include/CL/sycl/detail/pi.h   @smaslov-intel
-sycl/include/CL/sycl/detail/pi.hpp @smaslov-intel
-sycl/include/CL/sycl/detail/pi* @smaslov-intel
-sycl/plugins/ @smaslov-intel
-sycl/source/detail/pi.cpp @smaslov-intel
-sycl/source/detail/plugin.hpp @smaslov-intel
-sycl/source/detail/posix_pi.cpp @smaslov-intel
-sycl/source/detail/windows_pi.cpp @smaslov-intel
+# Level Zero plugin
+sycl/sycl/plugins/level_zero @intel/dpcpp-l0-pi-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:33]: @vladimirlaz, code review assignment is broken.
I think it's due to bugs in the .github/CODEOWNERS file.

sycl/sycl/plugins/level_zero -> sycl/plugins/level_zero?

File: .github/CODEOWNERS [hunk 11]
```diff
@@ -1,116 +1,55 @@
 * @bader
 
 # Front-end compiler
-clang/ @premanandrao @elizabethandrews @smanna12
+clang/ @intel/dpcpp-cfe-reviewers
 
-# Driver
-clang/**/Driver @mdtoguchi @AGindinson @hchilama
+# Clang driver
+clang/**/Driver @intel/dpcpp-clang-driver-reviewers
```
Comment (bader) [inline .github/CODEOWNERS:7]: Please, make sure that all these teams exist and have write access to the repository.

Comment (vladimirlaz) [reply .github/CODEOWNERS:7]: Good catch. Fixed for all teams both in intel/llvm and intel/llvm-test-suite.

Comment (bader) [reply .github/CODEOWNERS:7]: Okay. Please, test automatic assignment. I think it's still broken due to the issue in my previous comment.

Comment (vladimirlaz) [reply .github/CODEOWNERS:7]: it worked on change of the sources in the PR (fixed 2 hours form now). No automatic assignemnt for existent PR.

**Pull request merged**